resources/device_probe.py

```diff
@@ -880,6 +884,33 @@ def bluetooth_probe(self):
         elif any("Bluetooth" in usb_device.product_name for usb_device in self.usb_devices):
             self.bluetooth_chipset = "Generic"
 
+    def keyboard_probe(self):
+        if not self.usb_devices:
+            return
+
+        legacy_ids = [526,527,528,532,533,534,525,536,537,538,539,540,553,554,555] # These keyboards were stripped of their personalities after being dropped
+        modern_ids = [547,548,549,560,561,562,566,567,568,575,576,577,578,579,580,581,582,583,585,586,587,601,602,603,610,611,612,588,589,590,594,595,596] #These keyboards remained in AppleUSBTCKeyboard until the kext was removed
+        
+        for usb_device in self.usb_devices:
+            if usb_device.device_id in legacy_ids and usb_device.vendor_id == 1452:
+                self.keyboard_type = "Legacy"
+            elif usb_device.device_id in modern_ids and usb_device.vendor_id == 1452:
+                self.keyboard_type = "Modern"
+
+    def trackpad_probe(self):
+        if not self.usb_devices:
+            return
+        
+        legacy_ids = [526,527,528,778,779,532,533,534,535,536,537,538,539,540,553,554,555] # Overlap with keyboard IDs as some units have unified USB topcase devices
```

Comment:
If there's an overlap, maybe it's a better idea to combine the two functions? ie.

```
topcase probe ->
   if vendor matches:
      if device in [a, b, c]:
         # unified topcase
         self.trackpad_type = "Legacy"
         self.keyboard_type = "Legacy"
      elif device in [e, f, g]:
         self.trackpad_type = "Legacy"
      elif device in [g, h, I]:
         self.keyboard_type = "Legacy"
```
etc